src/sql-parser/src/ast/defs/query.rs

```diff
@@ -133,6 +135,7 @@ pub enum SetExpr<T: AstInfo> {
     },
     Values(Values<T>),
     // TODO: ANSI SQL supports `TABLE` here.
+    Show(Box<Statement<T>>),
```

Comment:
Can we factor out an intermediate `enum ShowStatement` that we can use here, instead of allowing any `Statement`? Otherwise you'll be able to manually construct a `SetExpr::Show` that's totally invalid.